### ISSUE TITLE
Is it possible to ExecuteJavaScript or emulate user actions like key pressing on loaded page?

### ISSUE THREAD

**Issue by ziomyslaw** [ElectronNET/Electron.NET#183]
I am looking for solution to replace NightmareJs with c# implementation. 
The NightmareJs wraps electron api to allows not only to type or click but execute js as well i.e.
[win.webContents.executeJavaScript(src)](https://github.com/segmentio/nightmare/blob/25467e5a1071c05dcbce7f58e137f89c4b8fdd93/lib/runner.js#L395)
`WebContents` does not provide such methods.
Is there any equivalent way to do this?

**Comment by kwakuduahc1** [comment]
Are you using an SPA framework or not?

**Comment by ziomyslaw** [comment]
I used to use NightmareJs + MeteorJs, but I'm looking for c# solution. Something faster than Selenium+Chrome.

**Comment by niklr** [comment]
I think what he means is similar to this question: https://stackoverflow.com/questions/28920621/how-to-call-a-javascript-function-on-a-web-page-rendered-by-electron

What is the equivalent to the following solution in Electron.NET?

```
app.on('ready', function() {

  ...

  mainWindow.webContents.on('did-finish-load', function() {
    mainWindow.webContents.executeJavaScript("alert('Hello There!');");
  });

  ...

});
```

**Comment by LittleEndu** [comment]
Is there any solution to this? Being able to execute javascript and to convert the resulting javascript object into a C# object would be very useful.

**Comment by GregorBiswanger** [comment]
@LittleEndu yes, that works with the **ElectronHost Hook feature**. Take a look at the example of the WebApp here. There is a sample code deposited. In the example application you can also get a step by step guide.

**Comment by MrCircuit** [comment]
I managed to inject the "electron-chromedriver" npm package into the build command of ElectronNET. It was simply by adding another line after npm install at line 123 in BuildCommand.cs:

                ProcessHelper.CmdExecute("npm install electron-chromedriver", tempPath);

Of course, this could also be achieved by editing the packages.json file.

Using this modified build command, I built the whole ElectronNet project. Then created a small .NET test project with ChromeDriver that uses the "BinaryLocation" property at driver creation. With this setup, I can control and test the ElectronNET application.
